src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +42,24 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(e) => setQuery(e.target.value)}
```

Comment:
The `setQuery` function is called on every key press. This might cause performance issues if the list of TODOs is large. Consider using a debounced version of the function to limit the number of dispatches.